### PR TITLE
Music cont out proxy

### PR DESCRIPTION
**Pull request by gtrensch**
This pull request follows up on PR #280 and replaces it.
`New neuron type to support forwarding continuous values via the MUSIC continuous output port.`

In addition (please see #280) the following has been added:
- Due to the delay on this issue the current master was much ahead so that a small fix for the DataLoggingRequest class was needed. (e1ce154)
- A SLI test for the music_cont_out_proxy / music_in_cont_proxy was added. (3a3ca79, be02c6e, d61e473)
- Due to the merge with the current master the CPPCHECK messages caused an overflow of the build log file on Travis which has a 4MB limit. Some logic was added to the build script to only report on the first 10 CPPCHECK messages per file. (0ce6021)

### CODE REVIEW

File: extras/static_code_analysis.sh [hunk 1]
```diff
@@ -165,9 +168,17 @@ for f in $FILE_NAMES; do
...
           do
             print_msg "MSGBLD0155: " "[CPPC] $line"
+            if $RUNS_ON_TRAVIS; then
+              msg_count+=1
+              if [ ${msg_count} -ge ${MAX_CPPCHECK_MSG_COUNT} ]; then 
+                print_msg "MSGBLD0156: " "[CPPC] MAX_CPPCHECK_MSG_COUNT (${MAX_CPPCHECK_MSG_COUNT}) reached for file: $f"
+                break
+              fi
+            fi
           done
```
Comment (jougs) [inline extras/static_code_analysis.sh]: I suggest the following code (without explicit counter) as a replacement for your lines 171-181:
```bash
head -n ${MAX_CPPCHECK_MSG_COUNT} ${f_base}_cppcheck.txt | while read line; do
  print_msg "MSGBLD0155: " "[CPPC] $line"
done

if ${RUNS_ON_TRAVIS} && [ `wc -l ${f_base}_cppcheck.txt` -ge ${MAX_CPPCHECK_MSG_COUNT} ]; then
  print_msg "MSGBLD0156: " "[CPPC] MAX_CPPCHECK_MSG_COUNT (${MAX_CPPCHECK_MSG_COUNT}) reached for file: $f"
fi
```

File: models/music_cont_out_proxy.cpp [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+    // The permutation index map, contains global_index[local_index]
+    MUSIC::PermutationIndex* music_perm_ind = new MUSIC::PermutationIndex(
+      &music_index_map.front(), music_index_map.size() );
+
+    MUSIC::ArrayData* dmap =
+      new MUSIC::ArrayData( static_cast< void* >( &( B_.data_.front() ) ),
+        MPI::DOUBLE,
+        music_perm_ind );
+
```
Comment (jougs) [inline models/music_cont_out_proxy.cpp]: Remove superfluous blank line.

File: models/music_cont_out_proxy.cpp [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+nest::music_cont_out_proxy::update( Time const& origin,
+  const long from,
+  const long )
+{
+  /* There is nothing to request during the first time slice.
+     For each subsequent slice, we collect all data generated during the
+     previous
+     slice if we are called at the beginning of the slice. Otherwise, we do
+     nothing.
+   */
```
Comment (jougs) [inline models/music_cont_out_proxy.cpp]: Fix formatting/line breaks of comment.

File: pynest/examples/music_cont_out_proxy_example/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Music_cont_out_proxy example
+
+## Requirements
++ MUSIC 1.1.15 or higher
++ NEST 2.10.0 or higher compiled with MPI and MUSIC
```
Comment (jougs) [inline pynest/examples/music_cont_out_proxy_example/README.md]: Change to 2.12.1, as that's the earliest possible where this code will be made available.

File: extras/static_code_analysis.sh (hunk 1 above)
Comment (gtrensch) [inline extras/static_code_analysis.sh]: Wouldn't it be always just report on the first MAX_CPPCHECK_MSG_COUNT number of CPPCHECK messages? I you for example do:
`./extras/check_code_style.sh --file=models/aeif_cond_alpha.cpp --perform-cppcheck=on`
You want to see all the messages.

File: models/music_cont_out_proxy.cpp (hunk 2 above)
Comment (gtrensch) [inline models/music_cont_out_proxy.cpp]: I will fix this. But just for the protocol, you already approved it :) and I just merged it into my branch.

File: pynest/examples/music_cont_out_proxy_example/README.md (hunk 4 above)
Comment (gtrensch) [inline pynest/examples/music_cont_out_proxy_example/README.md]: Good point, I will change it.

File: extras/static_code_analysis.sh (hunk 1 above)
Comment (jougs) [inline extras/static_code_analysis.sh]: You are right. Just ignore my comment then ;-)

File: models/music_cont_out_proxy.cpp (hunk 2 above)
Comment (jougs) [inline models/music_cont_out_proxy.cpp]: > Was kümmert mich mein Geschwätz von gestern, nichts hindert mich, weiser zu werden -- Konrad Adenauer

Comment (gtrensch) [inline models/music_cont_out_proxy.cpp]: Sehr schoen!

**Comment by gtrensch** [comment]
@jougs searched for additional blank lines and removed them. 

Concerning the truncated log in build 2040: Too much CPPCHECK messages is very likely not the reason. An overflow is usally marked by the following message: `The log length has exceeded the limit of 4 MB (this usually means that the test suite is raising the same exception over and over). `

The log just looks truncated for an unknown reason. This is strange.

File: pynest/examples/music_cont_out_proxy_example/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Music_cont_out_proxy example
+
+## Requirements
++ MUSIC 1.1.15 or higher
++ NEST 2.12.1 or higher compiled with MPI and MUSIC
```
Comment (stinebuu) [inline pynest/examples/music_cont_out_proxy_example/README.md]: Shouldn't this be NEST 2.14.0 now?

Comment (gtrensch) [reply pynest/examples/music_cont_out_proxy_example/README.md]: @stinebuu thanks indeed, this is 2.14.0 now. I am going to change this.

**Comment by jougs** [comment]
Many thanks for the hard work.

**Pull request merged**